common/test/resources/karma-browserstack-reconfig.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Provides an exportable function for configuring a Karma setup for execution in our CI processes
+ * via BrowserStack.
+ *
+ * @param {*} baseConfigParams   The project's base configuration parameters, independent of local vs CI test mode
+ * @param {*} targetSet          Either "partial" or "full":  "partial" aims for a smaller yet representative
+ *                               subset of target browser/device pairings.  Defaults to "full" if unspecified.
+ * @returns {(config) => void}   The function to be exported in the final Karma configuration file
+ */
+module.exports = function(baseConfigParams  /* the project's base configuration */,
+                          targetSet         /* a 'flag' for whether to use a large set of test targets */) {
+  var timeouts = baseConfigParams.client.args[0];
+  var browserStackModifier = 10;
+
+  for(key in timeouts) {
+    if(typeof timeouts[key] == 'number') {
+      timeouts[key] = timeouts[key] * browserStackModifier;
+    }
+  }
+
+  timeouts.mobileFactor = 2; // Extra timeout padding for running on a remote mobile device.
+
+  /*
+   * Definition of utility functions for managing our browser lists.
+   */
+  var mergeLaunchers = function() {
+    var mergedDefs = {};
+    var i;
+    for(i=0; i < arguments.length; i++) {
+      for(var name in arguments[i]) {
+        mergedDefs[name] = arguments[i][name];
+
+        // Necessary for Karma to process it properly.
+        mergedDefs[name].base = 'BrowserStack';
+      }
+    }
+
+    return mergedDefs;
+  }
+
+  var toBrowserList = function(mergedSet) {
+    var list = [];
+    for(var name in mergedSet) {
+      list.push(name);
+    }
+
+    return list;
+  }
+
+  var PRIMARY_LAUNCHERS = {
```

Comment:
Given these need to be updated periodically, is it possible to have these definitions somewhere less hidden away -- I wouldn't think to look for them in this file.

Perhaps if we move them to the top of the file, even outside the `module.exports`(?) and rename this file to `karma-browserstack-config.js`, and then a little comment to separate? 

```
// ------------------------------------
// end of configuration section
// ------------------------------------
```

Or something like that?